components/feature/sitepermissions/src/main/java/mozilla/components/feature/sitepermissions/SitePermissionsStorage.kt

```diff
@@ -51,6 +59,9 @@ class SitePermissionsStorage(
      * @param sitePermissions the sitePermissions to be updated.
      */
     fun update(sitePermissions: SitePermissions) {
+        coroutineScope.launch {
```

Comment:
Why does `clearData` need to run on main scope?

Reply:
It's because [GeckoEngine](https://github.com/mozilla-mobile/android-components/blob/b49254d96af0e8cf80f00be33b4dc198c2800af0/components/browser/engine-gecko-nightly/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngine.kt#L425) implements it, and its operations need have a looper. As `update` is an operation that will happens on worker thread.